### PR TITLE
ci: introduce v benchmarks inside the ci workflow

### PR DESCRIPTION
**Pull request by vincenzopalazzo**
This is one of many steps that bring inside the CI a complete benchmarks about V lang where will be possible to check if any feature introduced or bug fixed will introduce a decrease of performance with a gap that is accepted we will discuss when the checker is ready.

Signed-off-by: Vincenzo Palazzo <vincenzopalazzodev@gmail.com>

### CODE REVIEW

**Comment by JalonSolov** [comment]
https://fast.vlang.io is updated on every commit, so there is already a mechanism to tell if there were any speed changes.

Is this change better than what fast.vlang.io shows now?  Should they be merged?

**Comment by spytheman** [comment]
I admit, that I also do not understand fully the benefit myself, beside testing/benchmarking not only the compiler speed (what https://fast.vlang.io/ does), but other modules too.

That said, it does not hurt to try it imho, for a month or two, to see what stats are collected, and if they are useful/actionable.

**Comment by spytheman** [comment]
The cost is ~3min additional time on that CI JOB:
image , which given that we have jobs that take over an hour (the sanitized ones), is not much.

It seems to also upload a .zip file of < 10KB, with the produced .json files from the runs/results.

File: .github/workflows/ci_v_benchmark.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    name: Run
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+    - name: Set up GCC
+      uses: egor-tensin/setup-gcc@v1
+      with:
+        version: latest
+        platform: x64
```
Comment (spytheman) [inline .github/workflows/ci_v_benchmark.yml:22]: why is that needed?
Afaik, all the github runners already do have gcc installed.

Comment (vincenzopalazzo) [reply .github/workflows/ci_v_benchmark.yml:22]: I will remove it, and we check it thanks!

File: .github/workflows/ci_v_benchmark.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    - name: Setup cmake
+      uses: jwlawson/actions-setup-cmake@v1.12
+      with:
+        cmake-version: '3.16.x'
```
Comment (spytheman) [inline .github/workflows/ci_v_benchmark.yml:26]: same - afaik, cmake is installed by default

**Comment by vincenzopalazzo** [comment]
>It seems to also upload a .zip file of < 10KB, with the produced .json files from the runs/results.

Yes it contains the json file with the benchmark run result, in case we have performance regression

File: .github/workflows/ci_v_benchmark.yml (hunk 2 above)
Comment (spytheman) [reply .github/workflows/ci_v_benchmark.yml:26]: From https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-Readme.md :
```
CMake 3.23.0
GNU C++ 9.4.0, 10.3.0
```

Comment (vincenzopalazzo) [reply .github/workflows/ci_v_benchmark.yml]: Removed, we will give a try :) Thanks!
